### ISSUE TITLE
OkHttp engine ignores preconfigured OkHttpClient in 1.3.1

### ISSUE THREAD

**Issue by drymarev** [ktorio/ktor#1646]
**Ktor Version and Engine Used (client or server and name)**
version, client/server, additional ktor modules included
1.3.1, client, ktor-client-okhttp

**Describe the bug**
`OkHttp` engine ignores preconfigured `OkHttpClient` in 1.3.1. This works just fine with 1.3.0 though.

**To Reproduce**
1. Create an instance of `OkHttpClient` with `HttpLoggingInterceptor` with your favorite logger.
2. Create an instance of `HttpClient` with `OkHttp` engine that has the following configuration:
```kotlin
fun createHttpClient(client: OkHttpClient) = HttpClient(OkHttp) {
    engine { preconfigured = client }
}
```
3. Make some HTTP calls
4. Find that nothing was logged

**Expected behavior**
Use preconfigured `OkHttpClient` if set.

**Comment by dmitrievanthony** [comment]
Hi @drymarev, 

Thanks for reporting, it looks like a critical issue. I think I understood the root cause, so I prepared PR with the fix (https://github.com/ktorio/ktor/pull/1648).

**Comment by dmitrievanthony** [comment]
I merged the fix, so it will be available in `1.3.2`.
